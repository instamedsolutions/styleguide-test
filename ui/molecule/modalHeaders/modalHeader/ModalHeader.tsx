import Icon from "@ui/atoms/icons/icon/Icon";
import H2 from "@ui/atoms/texts/base/H2";
import React, { FC } from "react";

export type ModalHeaderProps = {
  title: string;
  onClose: () => void;
  className?: string
};

const ModalHeader:FC<ModalHeaderProps> = ({ title, onClose, className }) => (
  <div className={className}>
    <H2>{title}</H2>
    <button onClick={onClose}>
      <Icon icon='close' />
    </button>
  </div>
);

export default ModalHeader;