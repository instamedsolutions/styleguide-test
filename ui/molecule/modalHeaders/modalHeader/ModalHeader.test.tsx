import React from 'react';
import render from '@ui/testing/render';

import ModalHeader from './ModalHeader';

describe('molecule/modalHeaders/modalHeader', () => {
  it('render correctly with empty title', () => {
    const tree = render(<ModalHeader title='' onClose={() => null} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
