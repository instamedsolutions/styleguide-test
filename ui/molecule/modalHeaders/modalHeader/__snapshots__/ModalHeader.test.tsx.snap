// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecule/modalHeaders/modalHeader render correctly with empty title 1`] = `
<div>
  <h2
    className="text-2xl font-bold"
  >
    
  </h2>
  <button
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </button>
</div>
`;
